src/main/scala/me/rexim/morganey/Main.scala

```diff
@@ -29,37 +29,54 @@ object Main extends SignalHandler {
     }
   }
 
+  def exitRepl() = {
+    System.exit(0)
+  }
+
+  def handleLine(con: ConsoleReader)(globalContext: List[MorganeyBinding], line: String): Option[List[MorganeyBinding]] = {
+    import LambdaParser.{parse, replCommand}
+    val nodeParseResult = parse(replCommand, line)
+
+    if (nodeParseResult.successful) {
+      val node = nodeParseResult.get
+      val computation = evalOneNodeComputation(node)(globalContext)
+
+      awaitComputationResult(computation) match {
+        case Success(MorganeyEval(context, result)) =>
+          result.foreach(t => con.println(smartPrintTerm(t)))
+          Some(context)
+        case Failure(e) =>
+          con.println(e.getMessage)
+          None
+      }
+    } else {
+      con.println(nodeParseResult.toString)
+      None
+    }
+  }
+
   def startRepl() = {
     Signal.handle(new Signal("INT"), this)
 
+    val running = true
     var globalContext = List[MorganeyBinding]()
     val con = new ConsoleReader()
     con.setPrompt("λ> ")
 
-    while (true) {
-      val line = con.readLine().trim()
+    def line(): String = {
+      val rawLine = con.readLine()
+      if (rawLine eq null) null
+      else rawLine.trim()
+    }
```

Comment:
This is fine, but how about this?

```
def line(): String =
  Option(con.readLine())
    .map(_.trim)
    .getOrElse(null)
```

I think it's more scala-ish :) Or, if the point to only check for null result in the while-loop below we can make it even simplier:

```
def line(): String = Option(con.readLine()).map(_.trim)
```

And check for `None` instead of `null` in the loop.

What do you think?